nixos/modules/security/pam.nix

```
@@ -474,6 +515,10 @@ let
                " kwalletd=${pkgs.kdeFrameworks.kwallet.bin}/bin/kwalletd5")}
           ${optionalString (cfg.enableGnomeKeyring)
               "session optional ${pkgs.gnome3.gnome-keyring}/lib/security/pam_gnome_keyring.so auto_start"}
+          ${optionalString cfg.gnupg.enable
+              "session optional ${pkgs.pam_gnupg}/lib/security/pam_gnupg.so"
+              + optionalString (cfg.gnupg.enable && cfg.gnupg.noAutostart) " no-autostart"
```

If I read correctly, you are already inside the `optionalString` of `cfg.gnupg.enable`:

```suggestion
              + optionalString cfg.gnupg.noAutostart " no-autostart"
```

Or perhaps this might be more readable (at least it's more intuitive to me):

```nix
          ${optionalString cfg.gnupg.enable
             "session optional ${pkgs.pam_gnupg}/lib/security/pam_gnupg.so${
                optionalString cfg.gnupg.noAutostart " no-autostart"}"
          }
```

---

You're absolutely right! I've fixed this in the latest commit